azure/Guidelines.md

```diff
@@ -725,19 +727,28 @@ You MAY support preflight requests...
 Preflight requests?  Often supported for CORS but could be used for any potentially expensive request.  Consider "EXPECT: 100-continue" for other requests that will return 100 Continue if the conditions are satisfactory or 417 Expectation Failed otherwise. Useful for conditionaly requests with large payloads. You should return the same error code that the service should use. 
 SHOULD provide documentation on what preflight checks will be validated.
 
-AWS uses this for S3 API
-
 
 -	Status codes
-o	GET: if the comparison fails, return 304 Not Modified (consider also returning Expires/Cache-Control/Age headers for caching scenarios)
-o	PUT/POST/DELETE: if the comparison fails, return 412 Precondition Not Met
-o	Consider forcing conditional headers on resource mutation, then use 428 Precondition Required if they're not present.
+* GET: if the comparison fails, return 304 Not Modified (consider also returning Expires/Cache-Control/Age headers for caching scenarios)
+* PUT/POST/DELETE: if the comparison fails, return 412 Precondition Not Met
+* Consider forcing conditional headers on resource mutation, then use 428 Precondition Required if they're not present.
 
 -	If-Match header - should support multiple values that are Or-ed together per HTTP/1.1.
 
 
+### Distributed Tracing & Telemetry
+Azure Core guidelines specify that you support telemetry data through both the ```User-Agent``` header, ```X-MS-UserAgent``` header, and Open Telemetry. 
+When a client library is created, code to handle telemetry and distributed tracing information will be automatically added. Telemetry information is vital to the effective operation of your service and should be a consideration from the outset of design and implementation efforts.
```

Comment:
Is that true across languages?  The default pipeline does this automatically in languages that have a default pipeline, but I think languages like Java would require you to include this yourself.

Reply:
Are there any other languages besides Java where this is not added automatically? I can update this wording a bit to reflect this. Also, and just to confirm we are on the same page, the fact the tooling does not handle this OOTB, does not change/alter the guidance of supporting both of headers.

Reply:
We'd have to loop in folks from JS/Python/C++/Go/etc. to confirm as this is a lower level detail.  What if we just changed the wording and leave what clients do/don't do by default up to the language specific client guidelines?

```suggestion
Client libraries are required to send telemetry and distributed tracing information on every  request. Telemetry information is vital to the effective operation of your service and should be a consideration from the outset of design and implementation efforts.
```

Reply:
Yep. I like this. I'll commit this change.